D-lama-service/d-lama-service/DataProcessing/DataSetReader.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Text;
+using System.Text.Json;
+
+namespace d_lama_service.DataProcessing
+{
+    /// <summary>
+    /// Reads textual files and returns data points in a list.
+    /// </summary>
+    public class DataSetReader
+    {
+        private readonly string[] permittedExtensions = { ".csv", ".json", ".txt" };
+        private readonly Encoding encoding = Encoding.UTF8;
+
+
+        public DataSetReader() { }
+
+        /// <summary>
+        /// Checks id the given file is supported.
+        /// </summary>
+        /// <param name="file"> The project ID. </param>
+        /// <returns> True if file extension is supported, else False. </returns>
+        public bool IsValidFormat(IFormFile file)
+        {
+            string fileExt = GetFileExtension(file);
+            if (string.IsNullOrEmpty(fileExt) || !permittedExtensions.Contains(fileExt))
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Checks id the given file is supported.
+        /// </summary>
+        /// <param name="file"> The project ID. </param>
+        /// <returns> A List of data point strings or an empty list. </returns>
+        public async Task<ICollection<string>> ReadFileAsync(IFormFile file)
+        {
+            ICollection<string> dataPoints = new List<string>();
+
+            if (file.Length <= 0 )
+            { 
+                return dataPoints;
+            }
+
+            var reader = new StreamReader(file.OpenReadStream(), encoding);
+
+            // read txt; DataPoints separated by new line
+            if (GetFileExtension(file) == ".txt" || GetFileExtension(file) == ".csv")
```

Comment:
I would change that so that it matches the Open-Closed principle. =>e.g. in the GetFileExtension method you will create a JsonDataParser / TxtDataParser which both are extracted in one interface (maybe IDataParser). With that, you can then easily just call the implementations Parse method and are welcome to new changes according to the open-closed principle

Reply:
Thank you for the input. I tried to apply your suggestion to the DataSetReader with my last commit, please have a look again, grazie!